### PR TITLE
Store experiments in contexts

### PR DESCRIPTION
**Pull request by DrJosh9000**
The main change is to associate experiments with contexts (as values). Enabling or disabling an experiment creates a new context.

Most code has context plumbed through correctly, so adopting it wasn't too hard. The bulk of the updates are to tests.

But now that experiments is not a "global variable", we can parallelise more tests!

### CODE REVIEW

**Comment by DrJosh9000** [comment]
This is ready, if anyone wants to poke at it.

**Review by triarius — COMMENTED** [review]
This makes me so happy!

Just mainly, I think the move to internal can happen now. It's supposedly imported by 2 libraries, but that just looks like a fork of the agent.

Event if it's imported by real modules, there's legitimate reason modules outside the agent to use it.

File: clicommand/global.go [hunk 1]
```diff
@@ -287,10 +293,11 @@ func withConfigFilePaths(paths []string) func(*cliconfig.Loader) {
 // future to clean up other resources. Importantly, the calling code does not
 // need to know or care about what the returned function does, only that it
 // must defer it.
-func setupLoggerAndConfig[T any](c *cli.Context, opts ...configOpts) (
+func setupLoggerAndConfig[T any](ctx context.Context, c *cli.Context, opts ...configOpts) (
 	cfg T,
 	l logger.Logger,
 	f *cliconfig.File,
+	newctx context.Context,
```
[the remark this answers is not in the document]
Comment (DrJosh9000) [reply clicommand/global.go]: Done and done.

File: experiments/experiments.go
[the remark this answers is not in the document]
Comment (DrJosh9000) [reply experiments/experiments.go:1]: Agreed, done.

**Review by triarius — APPROVED** [review]
💜

**Pull request merged**